pkg/object/object.go

```diff
@@ -227,9 +284,82 @@ func (v *Volume) validate() error {
 	return nil
 }
 
+func (sd *SecretData) validate() error {
+	var count int
+	if sd.Plaintext != nil {
+		count++
+	}
+	if sd.Base64 != nil {
+		count++
+	}
+	if sd.File != nil {
+		count++
+	}
+
+	switch count {
```

Comment:
`switch` just seemed a bit idiomatic to me, but meh, just personal preference.
I have added the `default` :+1: